maven-model-builder/src/test/java/org/apache/maven/model/path/DefaultUrlNormalizerTest.java

```diff
@@ -82,4 +82,8 @@ public void testPreservationOfDoubleSlashes()
                       normalize( "[fetch=]http://server.org/[push=]ssh://server.org/" ) );
     }
 
+    public void testNormalizeInputWithNoParentDirectoriesToAscendTo()
+    {
+        assertEquals("a/../", normalize("a/../"));
+    }
```

Comment:
Maybe the implementation is wrong, but comparing this with Path this is the result:

        assertEquals( "a/../",  Paths.get("a/../").normalize().toString() ); // fails
        assertEquals( "",  Paths.get("a/../").normalize().toString() ); // succeeds

The Path behavior is also what I would expect.

Reply:
@rfscholte So our impl is wrong?

Reply:
I'm not sure it's worth changing the behaviour; this class is used with absolute URLs (or paths), so this is very much an edge case.

However, I think it's (slightly) valuable to test that this case doesn't cause an error -- perhaps a different test name could make that clear?

Reply:
If we normalize URLs, the test for relative paths must fail. URLs are absolute.

Reply:
Thinking again; there's no reason to retain the existing behaviour if it's wrong. The change to match `Path` is simple enough, and doesn't break the existing tests: I've pushed it.

Ideally, the interface would be clearer about its contract, but this is the least surprising behaviour.